src/Functional.CQS.AOP.IoC.SimpleInjector.MetricsCapturing.Tests/ConventionBasedDecoratorRegistrationGatewayExtensionsTests.cs

```diff
@@ -0,0 +1,273 @@
+		{
+			foreach (var type in TestUtility.CQSHandlerContractTypes)
+				container.GetInstance(type).Should().BeOfType(MetricsCapturingTestUtility.UniversalMetricsCapturingDecoratorTypeLookupByCQSHandlerContractType[type]);
+		}
+
+		[Theory, SingletonRegistrationWithAllDecorationsEnabledArrangement]
+		public void ShouldApplyDecoratorIfConfigurationHasAllDecorationsEnabled_SingletonRegistration(Container container) => ShouldApplyDecoratorIfConfigurationHasAllDecorationsEnabled(container);
+
+		[Theory, TransientRegistrationWithAllDecorationsEnabledArrangement]
+		public void ShouldApplyDecoratorIfConfigurationHasAllDecorationsEnabled_TransientRegistration(Container container) => ShouldApplyDecoratorIfConfigurationHasAllDecorationsEnabled(container);
+
+		private static void ShouldApplyDecoratorIfConfigurationHasAllDecorationsEnabled(Container container)
+		{
+			foreach (var type in TestUtility.CQSHandlerContractTypes)
+				container.GetInstance(type).Should().BeOfType(MetricsCapturingTestUtility.UniversalMetricsCapturingDecoratorTypeLookupByCQSHandlerContractType[type]);
+		}
+
+		[Theory, SingletonRegistrationWithCommandAndQuerySpecificDecorationsEnabledButNoStrategyImplementationsArrangement]
+		public void ShouldNotApplyDecoratorWhenNoStrategyImplementationExists_SingletonRegistration(Container container) => ShouldNotApplyDecoratorWhenNoStrategyImplementationExists(container);
+
+		[Theory, TransientRegistrationWithCommandAndQuerySpecificDecorationsEnabledButNoStrategyImplementationsArrangement]
+		public void ShouldNotApplyDecoratorWhenNoStrategyImplementationExists_TransientRegistration(Container container) => ShouldNotApplyDecoratorWhenNoStrategyImplementationExists(container);
+
+		private static void ShouldNotApplyDecoratorWhenNoStrategyImplementationExists(Container container)
+		{
+			foreach (var type in TestUtility.CQSHandlerContractTypes)
+				container.GetInstance(type).Should().BeOfType(TestUtility.ImplementationTypeLookupByCQSHandlerContractType[type]);
+		}
+
+		[Theory, SingletonRegistrationWithAllDecorationsEnabledButNoStrategyImplementationsArrangement]
+		public void ShouldNotApplyAnyDecoratorsIfConfigurationHasAllDecorationsEnabledButNoStrategyImplementationsExist_SingletonRegistration(Container container) => ShouldNotApplyAnyDecoratorsIfConfigurationHasAllDecorationsEnabledButNoStrategyImplementationsExist(container);
+
+		[Theory, TransientRegistrationWithAllDecorationsEnabledButNoStrategyImplementationsArrangement]
+		public void ShouldNotApplyAnyDecoratorsIfConfigurationHasAllDecorationsEnabledButNoStrategyImplementationsExist_TransientRegistration(Container container) => ShouldNotApplyAnyDecoratorsIfConfigurationHasAllDecorationsEnabledButNoStrategyImplementationsExist(container);
+
+		private static void ShouldNotApplyAnyDecoratorsIfConfigurationHasAllDecorationsEnabledButNoStrategyImplementationsExist(Container container)
```

Comment:
For completeness, the only test case that is now missing here is
ShouldNotApplyAnyDecoratorsIfConfigurationHasUniversalDecorationsEnabledButNoStrategyImplementationsExist (both Singleton and Transient registration).

Just like `ShouldNotApplyDecoratorWhenNoStrategyImplementationExists` (which covers when Command/Query-specific decorations are enabled) and `ShouldNotApplyAnyDecoratorsIfConfigurationHasAllDecorationsEnabledButNoStrategyImplementationsExist` (which covers when all decorations are enabled), this case would assert that the container returns values of type `TestUtility.ImplementationTypeLookupByCQSHandlerContractType[type]`.

Reply:
I deleted one set of tests so there is now just one (which covers all three already anyway)